common/urlbuilder/url.go

```diff
@@ -17,6 +19,7 @@ var ErrInvalidURL = errors.New("URL format is incorrect")
 type URL struct {
 	delegate           *url.URL
 	queryParams        url.Values
+	unEncodeParams     map[string]struct{}
```

Comment:
Can you explain more about what there are changes to this file? This is shared by all connectors, so any changes here are highly risky. Can you explain the need to encode params? Please link to any relevant documentation.